### DISCUSSION TITLE
installing Composer 2 to ubuntu fails

### DISCUSSION THREAD

**Discussion by brianronnow** [composer/composer#9743] [Q&A / Support]
Hi there
First i am pretty new at this with composer.
I am trying to get composer to use at my Laravel project.
I have read the instructions
And when i test for the version it came with this: 
brian@brian-ubuntu:~/code/laravel$ composer --version
Composer 1.10.1 2020-03-13 20:34:27
I have used this code to install:
hp -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
php -r "if (hash_file('sha384', 'composer-setup.php') === '756890a4488ce9024fc62c56153228907f1545c228516cbf63f885e036d37e9a59d27d63f46af1d4d07ee0f76181c7d3') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
php composer-setup.php
php -r "unlink('composer-setup.php');"

What do i do wrong? 
Thank you.

**Comment by xabbuh** [comment]
Did you by chance also install Composer through Ubuntu's package manager?

**Reply by pgrimaud** [reply]
I guess yes, versions are the same https://packages.ubuntu.com/focal/composer.

Maybe he could just run `composer selfupdate --2`?

**Reply by brianronnow** [reply]
@xabbuh  No i have used the code 
@pgrimaud I will try your selfupdate..  and i get this: 
composer selfupdate --2
   Command "selfupdate" is not defined.

**Reply by brianronnow** [reply]
Now i have tried again and get this when i have installed it before used the php -r "unlink('composer-setup.php');"
![billede](https://user-images.githubusercontent.com/1021425/109788036-24e0d480-7c0f-11eb-8ccd-197ae61ee6eb.png)

brian@brian-ubuntu:~/code/laravel$ php composer-setup.php
All settings correct for using Composer
Downloading...

Composer (version 2.0.11) successfully installed to: /home/brian/code/laravel/composer.phar
Use it: php composer.phar

**Reply by brianronnow** [reply]
Now i have tried to getting further, but something does not work correct

brian@brian-ubuntu:~/code/laravel$ php -r "unlink('composer-setup.php');"
brian@brian-ubuntu:~/code/laravel$ php composer.phar update --with vendor/package:2.0.11
Composer could not find a composer.json file in /home/brian/code/laravel
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section

**Reply by pgrimaud** [reply]
I think you did it on the wrong way.

**First**, you should remove composer installed with unbuntu packages : 

`sudo apt remove composer`

**Next**, move to a new folder on CLI and download composer : https://getcomposer.org/download/ 

**Then**, install the binary **globally** on your computer : https://getcomposer.org/doc/00-intro.md#globally

You should do this on the folder you downloaded composer.phar file `mv composer.phar /usr/local/bin/composer`

**Finally**, run the command `composer --version`; you should see : `Composer version 2.0.11 2021-02-24 14:57:23`
